### PR TITLE
Adapt summary function to circular variables 

### PR DESCRIPTION
**Pull request by agustinaarroyuelo**
## Description

In this PR I changed the `az.summary` function to display statistics for circular variables more neatly. It is related to issue #339. 

`torsionals = az.load_arviz_data('glycan_torsion_angles')`
`az.summary(torsionals, circ_var_names=['tors'])` 

outputs:

sc_summary

## Checklist
- [X] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [X] New features are properly documented (with an example if appropriate)?
- [X] Includes new or updated tests to cover the new feature
- [X] Code style  correct (follows pylint and black guidelines)
- [x] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: arviz/stats/stats.py [hunk 1]
```diff
@@ -992,7 +992,7 @@ def summary(
     fmt: str = "wide",
     kind: str = "all",
     round_to=None,
-    include_circ=None,
+    circ_var_names=None,
```
Comment (OriolAbril) [inline arviz/stats/stats.py:995]: I think we should keep `include_circ` for now and raise a deprecation warning when used to ease transition

**Comment by agustinaarroyuelo** [comment]
Thanks for your comments @OriolAbril and @ahartikainen. I updated the PR description. As you can see, circular statistics are placed only where the circular variables are, so no need for extra columns. 

Maybe waiting for the new example to update the doctrings will be better but I will check with @aloctavodia

**Comment by ahartikainen** [comment]
That looks great.

This PR doesn't need to address this, but should we add a column for `is_circular` and use True for variables that are?
Normal summary doesn't need this, but maybe when we use circular variables it might be a good idea.

Also (I forgot already this) are those in degrees or in radians?

File: arviz/stats/stats.py [hunk 2]
```diff
@@ -1258,17 +1265,16 @@ def summary(
...
-                "circular_hdi_{:g}%".format(100 * (1 - alpha / 2)),
-                "circular_mcse",
-            )
-        )
+
+    if circ_var_names:
+
+        if kind != "diagnostics":
+            for metric, circ_stat in zip(
+                metrics[:5], (circ_mean, circ_sd, circ_hdi_lower, circ_hdi_higher, circ_mcse)
```
Comment (canyon289) [inline arviz/stats/stats.py:1273]: Whats 5 here? Would add comment for the magic number

Comment (aloctavodia) [reply arviz/stats/stats.py]: Agree about the comment. 
The 5 is related to the "stats" columns, the diagnostics are computed the same for linear and circular variables. Actually, we have to think if the diagnostics also need to be changed for circular variables (not part of this PR).

**Comment by aloctavodia** [comment]
Not sure about having a "true column" for circular variables. I think a better approach is to have that information included in the InferenceData object.

**Comment by agustinaarroyuelo** [comment]
> That looks great.
> 
> This PR doesn't need to address this, but should we add a column for `is_circular` and use True for variables that are?
> Normal summary doesn't need this, but maybe when we use circular variables it might be a good idea.
> 
> Also (I forgot already this) are those in degrees or in radians?

This angles are in radians

**Pull request merged**